ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -340,10 +312,61 @@ def create_random_action(self, n_agents: int) -> np.ndarray:
                         size=(n_agents),
                         dtype=np.int32,
                     )
-                    for i in range(self.action_size)
+                    for i in range(self.discrete_size)
                 ]
             )
-            return action
+        return action
+
+    def validate_action(
+        self, actions: np.ndarray, n_agents: int, name: str
+    ) -> np.ndarray:
+        """
+        Validates that action has the correct action dim
+        for the correct number of agents and ensures the type.
+        """
+        if self.continuous_size > 0:
+            _size = self.continuous_size
+        else:
+            _size = self.discrete_size
+        _expected_shape = (n_agents, _size)
+        if actions.shape != _expected_shape:
+            raise UnityActionException(
+                f"The behavior {name} needs an input of dimension "
+                f"{_expected_shape} for (<number of agents>, <action size>) but "
+                f"received input of dimension {actions.shape}"
+            )
+        _expected_type = np.float32 if self.is_continuous() else np.int32
+        if actions.dtype != _expected_type:
+            actions = actions.astype(_expected_type)
+        return actions
+
+    @staticmethod
+    def make_continuous(continuous_size: int) -> "ActionSpec":
```

Comment:
Not sure if make is the best word here. `make` sounds to me like we are making an ActionSpec into another. I think `new_continuous` or `create_continuous` is more appropriate (I might be wrong on the english here)

Reply:
This is to be consistent with the MakeContinuous and MakeDiscrete on C#

Reply:
Do we prefer consistency over grammatical correctness ? If we do, ignore this comment.

Reply:
I'm with @vincentpierre on this - consistency is nice in general, but not if the other is badly named.

Reply:
I totally get your point but I tend to lean toward being consistent. Maybe we should think about renaming C#/Python mirror entities like ActionBuffers etc. @chriselion @ervteng

Reply:
We're limited in what we can rename in C#; we can deprecate old methods and add new ones more to our liking. Classes and structs are harder though.

Reply:
renamed to create_x